lesson18/app.js

```diff
@@ -1,213 +1,193 @@
-const url = "https://jsondata.okiba.me/v1/json/ngovH210301060343",
-  parent = document.getElementById("js-parent"),
-  imagesFragment = document.createDocumentFragment();
-
-let interval;
-let imageLists;
-let bottons;
-
-function getImageLists(target) {
-  imageLists = [...document.querySelectorAll(target)];
-  return imageLists;
-}
-function getButtons(target) {
-  buttons = [...document.querySelectorAll(target)];
-  return buttons;
-}
+const parent = document.getElementById("js-parent");
 
-const wait = (sec) => {
-  return new Promise((resolve) => {
-    setTimeout(resolve, sec * 1000);
-  });
+const sliderStatus = {
+  count: 0,
+  currentPagination: "",
+  imageLists: "",
+  interval: "",
 };
 
-async function getJsonData() {
-  try {
-    await wait(3);
-    const response = await fetch(url),
-      jsonData = await response.json();
-    return jsonData;
-  } catch (err) {
-    throw Error(err);
-  } finally {
-    console.log("終了しました");
-  }
-}
+const renderSlider = ({ data }) => {
+  parent.appendChild(getImageList(data));
+  parent.after(getDirectionButtons());
+  parent.after(getPaginationText());
+  parent.after(getPaginationButtonContainer());
+};
 
-async function init() {
-  let data;
-  try {
-    data = await getJsonData();
-  } catch (e) {
-    console.error();
-  } finally {
-    console.log("終了しました");
-  }
-  if (data.length !== 0) {
-    createElements(data);
-  } else {
-    parent.innerHTML = "data is empty";
-  }
-}
-init();
+const autoPlay = {
+  start: function () {
+    sliderStatus.interval = setInterval(() => {
+      const isActive = document.querySelector(".is-active");
+      isActive.classList.remove("is-active");
+      sliderStatus.count =
+        sliderStatus.count < sliderStatus.imageLists.length - 1
+          ? ++sliderStatus.count
+          : 0;
+      sliderStatus.imageLists[sliderStatus.count].classList.add("is-active");
+      controlSlidersMovement();
+    }, 3000);
+  },
+  reset: function () {
+    clearInterval(sliderStatus.interval);
+    this.start();
+  },
+};
 
-function createElements(data) {
-  createOfListImageItem(data.images);
-  createArrowButton();
-  createPagination();
-  createButtonPagination();
-  getButtons(".button-pagination");
-  autoPagenation();
-}
-
-function createOfListImageItem(items) {
-  items.forEach((item, index) => {
-    const ListImageItemImg = document.createElement("img"),
-      ListImageItem = document.createElement("li");
-    ListImageItem.id = index;
+const getImageList = (imagesData) => {
```

Comment:
返している名前と同じにするともっといいです
```suggestion
const getImagesFragment = (imagesData) => {
```

Reply:
Thank you! Correction made.
[0d5be49](https://github.com/443n0511/js-lesson/pull/36/commits/0d5be4987ae33d2f5192e0b2139ca6c9298e5c42)